### ISSUE TITLE
Can't reproduce environment from .yml file

### ISSUE THREAD

**Issue by kalefranz** [conda/conda#3369]
_From @eco32i on August 19, 2016 15:14_

I am trying to create an environment from this `.yml` file: https://gist.github.com/eco32i/206d2f53ff5f9412b2c7a971129ac540

But for some reason it fails to install packages from `bioconda` channel:

```
$ conda env create -f environment.yml

Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /home/ilya/miniconda3:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    decorator-4.0.10           |           py35_0          12 KB
    ipython_genutils-0.1.0     |           py35_0          33 KB
    jsonschema-2.5.1           |           py35_0          56 KB
    traitlets-4.2.2            |           py35_0         113 KB
    jupyter_core-4.1.1         |           py35_0          51 KB
    nbformat-4.0.1             |           py35_0         113 KB
    ------------------------------------------------------------
                                           Total:         379 KB

The following NEW packages will be INSTALLED:

    decorator:        4.0.10-py35_0
    ipython_genutils: 0.1.0-py35_0 
    jsonschema:       2.5.1-py35_0 
    jupyter_core:     4.1.1-py35_0 
    nbformat:         4.0.1-py35_0 
    traitlets:        4.2.2-py35_0 

Proceed ([y]/n)? yes

Fetching packages ...
decorator-4.0. 100% |#########################################| Time: 0:00:00   7.15 MB/s
ipython_genuti 100% |#########################################| Time: 0:00:00   1.31 MB/s
jsonschema-2.5 100% |#########################################| Time: 0:00:00   1.66 MB/s
traitlets-4.2. 100% |#########################################| Time: 0:00:00   1.01 MB/s
jupyter_core-4 100% |#########################################| Time: 0:00:00   1.46 MB/s
nbformat-4.0.1 100% |#########################################| Time: 0:00:00   1.85 MB/s
Extracting packages ...
[      COMPLETE      ]|############################################################| 100%
Linking packages ...
[      COMPLETE      ]|############################################################| 100%
Fetching package metadata .........
Solving package specifications: .
Error: Packages missing in current linux-64 channels: 
  - bioconda::bcftools 1.3.1 0
  - bioconda::curl 7.45.0 1
  - bioconda::htslib 1.3.1 0
  - bioconda::libxml2 2.9.3 0
  - bioconda::pybedtools 0.7.8 py35_1
  - bioconda::pysam 0.9.1 py35_0
  - bioconda::python-dateutil 2.3 py35_0
  - bioconda::samtools 1.3.1 2
```

This fails on both MacOSX and Ubuntu 16.04. The file was created by running  `conda env export > environment.yml` from a working `conda` environment.

The output of `conda info --all`:

```
Current conda install:

             platform : linux-64
        conda version : 4.1.11
    conda-env version : 2.5.2
  conda-build version : not installed
       python version : 3.5.2.final.0
     requests version : 2.10.0
     root environment : /home/ilya/miniconda3  (writable)
  default environment : /home/ilya/miniconda3
     envs directories : /home/ilya/miniconda3/envs
        package cache : /home/ilya/miniconda3/pkgs
         channel URLs : https://conda.anaconda.org/bioconda/linux-64/
                        https://conda.anaconda.org/bioconda/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/ilya/.condarc
         offline mode : False
    is foreign system : False

# conda environments:
#
root                  *  /home/ilya/miniconda3

sys.version: 3.5.2 |Continuum Analytics, Inc.| (defau...
sys.prefix: /home/ilya/miniconda3
sys.executable: /home/ilya/miniconda3/bin/python3
conda location: /home/ilya/miniconda3/lib/python3.5/site-packages/conda
conda-build: None
conda-env: /home/ilya/miniconda3/bin/conda-env
user site dirs: ~/.local/lib/python3.4

CIO_TEST: <not set>
CONDA_DEFAULT_ENV: <not set>
CONDA_ENVS_PATH: <not set>
LD_LIBRARY_PATH: <not set>
PATH: /home/ilya/miniconda3/bin:/home/ilya/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ilya/bin/sratoolkit/bin:/home/ilya/bin/sratoolkit/bin
PYTHONHOME: <not set>
PYTHONPATH: <not set>

WARNING: could not import _license.show_info
# try:
# $ conda install -n root _license
```

And `conda list conda` shows this:

```
# packages in environment at /home/ilya/miniconda3:
#
conda                     4.1.11                   py35_0  
conda-env                 2.5.2                    py35_0
```
Does it have anything to do with the recommendation to create `environment.yml` by hand? Should I only include "top-level" packages and let `conda` figure out dependencies by itself? Or am I missing something obvious?

_Copied from original issue: conda/conda-env#269_

**Comment by kalefranz** [comment]
_From @oyse on August 30, 2016 7:34_

This looks like exactly the same problem I am having.

conda-env does not appear to be working correctly with the channel prefix "bioconda" or in my case and internal channel on our corporate server.

The problem seems to be in how it matches packages against the name.

I made some changes to resolve.py in the conda package to get it to work, but this needs a proper fix. I have uploaded the changed file here: https://gist.github.com/oyse/b8e8cf43ffd8a4910185fcd4e44dafd5
